### PR TITLE
[CALCITE-5651] Inferred scale for decimal should not exceed maximum allowed scale

### PR DESCRIPTION
**Pull request by mbudiu-vmw**
This PR includes the fix in #3155 as well, because it depends on it.

### CODE REVIEW

**Comment by rubenada** [comment]
@mbudiu-vmw I think the PR makes sense. There are some style errors though, could you please take care of them?

File: core/src/main/java/org/apache/calcite/sql/type/SqlTypeUtil.java [hunk 1]
```diff
@@ -1776,8 +1776,9 @@ public static boolean isAtomic(RelDataType type) {
    * type system. */
   public static RelDataType getMaxPrecisionScaleDecimal(RelDataTypeFactory factory) {
     int maxPrecision = factory.getTypeSystem().getMaxNumericPrecision();
+    int maxScale = factory.getTypeSystem().getMaxNumericScale();
     // scale should not greater than precision.
-    int scale = maxPrecision / 2;
+    int scale = Math.min(maxPrecision / 2, maxScale);
```
Comment (mbudiu-vmw) [inline core/src/main/java/org/apache/calcite/sql/type/SqlTypeUtil.java:1781]: Personally I think that the choice of maxPrecision/2 is rather arbitrary, but I didn't want to change too many things at once. This used to be maxScale, and that should always be smaller than maxPrecision.

**Comment by zabetak** [comment]
@mbudiu-vmw Small tip: it's better to avoid force-pushes after a review has started cause it makes review history hard to follow; no big deal for this change but it can be a pain in larger changes.

**Comment by rubenada** [comment]
@zabetak this PR needs [CALCITE-5650](https://issues.apache.org/jira/browse/CALCITE-5650); if I understand correctly, you plan to take care of that, so I'll pause the merge of the current PR until then, ok?

**Comment by zabetak** [comment]
@rubenada Sounds good let's do that!

**Comment by rubenada** [comment]
[CALCITE-5650](https://issues.apache.org/jira/browse/CALCITE-5650); has been merged (thanks @zabetak); @mbudiu-vmw  could you please revisit this PR?

**Comment by mbudiu-vmw** [comment]
I have rebased. How does one run the style checkers locally before submission?
I couldn't find this in the https://calcite.apache.org/develop/#contributing guidelines.

**Comment by rubenada** [comment]
> I have rebased. How does one run the style checkers locally before submission? I couldn't find this in the https://calcite.apache.org/develop/#contributing guidelines.

If I am not mistaken a local `gradlew build`  should include the verification of the code style

**Comment by rubenada** [comment]
LGTM.
Thanks @mbudiu-vmw  for rebasing. Could you please amend the commit message to simply its ticket number and title: `[CALCITE-5651] Inferred scale for decimal should not exceed maximum allowed scale` ?

**Comment by mbudiu-vmw** [comment]
Changed the commit message.
